NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
+                                                        config=self._config,
+                                                        time_logger=self.__time_logger,
+                                                        min_efield_amplitude=float(self._config['speedup']['min_efield_amplitude']) * self._Vrms_efield_per_channel[sid][channel_id],
+                                                        distance_cut=self._get_distance_cut)
+                    else:
+                        sim_station = calculate_sim_efield_for_emitter(emitters=event_group.get_sim_emitters(),
+                                            sid=sid, cid=channel_id,
+                                            det=self._det, propagator=self._propagator, medium=self._ice, config=self._config,
+                                            rnd=self._rnd, antenna_pattern_provider=self._antenna_pattern_provider,
+                                            min_efield_amplitude=float(self._config['speedup']['min_efield_amplitude']) * self._Vrms_efield_per_channel[sid][channel_id],
+                                            time_logger=self.__time_logger)
+                    if sim_station.is_candidate():
+                        candidate_station = True
+                    # skip to next channel if the efield is below the speed cut
+                    if len(sim_station.get_electric_fields()) == 0:
+                        logger.info(f"Eventgroup {event_group.get_run_number()} Station {sid} channel {channel_id:02d} has {len(sim_station.get_electric_fields())} efields, skipping to next channel")
                         continue
 
-                    # for special cases where only EM or HAD showers are simulated, skip all events that don't fulfill this criterion
-                    if self._cfg['signal']['shower_type'] == "em":
-                        if self._fin['shower_type'][self._shower_index] != "em":
-                            continue
-                    if self._cfg['signal']['shower_type'] == "had":
-                        if self._fin['shower_type'][self._shower_index] != "had":
-                            continue
+                    # applies the detector response to the electric fields (the antennas are defined
+                    # in the json detector description file)
+                    apply_det_response_sim(sim_station, self._det, self._config, self.detector_simulation_filter_amp,
+                                           event_time=self._evt_time, time_logger=self.__time_logger,
+                                           detector_simulation_part1=self.detector_simulation_part1)
+                    logger.debug(f"adding sim_station to station {sid} for event group {event_group.get_run_number()}, channel {channel_id}")
+                    station.add_sim_station(sim_station)  # this will add the channels and efields to the existing sim_station object
```

Comment:
I do not really like this "interface". It is vert unintuitive and actually needed changes in the framework. Why don't we return a list of efields from `calculate_sim_efield` and add it to the sim_station without creating this "dummy" sim_station objects?

Reply:
I first implemented it that way but I ran into some problems with that. 
E.g.  we need additional meta attributes such as "event time" and if it is an "inince" shower or a "cosmic ray". The sim_station is the natural place to store these. 
In addition, this is exactly how the NuRadio event structure is set up. The `sim_station` is the container object of simulated electric field.

Reply:
Both functions (`calculate_sim_efield*`) set `sim_station.set_is_neutrino()` and `sim_station.set_candidate(True or False)`. The latter could be done later? What am I missing?

If it is necessary to return a sim_station can we rename `add_sim_station` to `merge_sim_station`?